jax/_src/compiler.py

```diff
@@ -287,7 +287,8 @@ def compile_or_get_cached(
 
   try:
     cache_key = compilation_cache.get_cache_key(
-        computation, devices, compile_options, backend)
+        computation, devices, compile_options, backend, 
+        config.remove_custom_partitioning_ptr_from_cache_key.value)
```

Comment:
why pass this as an argument? You can just read that value in the function right?